metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/data_resolver.py

```diff
@@ -0,0 +1,660 @@
+            logger.warning(
+                "Failed to generate the PowerBi access token. Please check input configuration"
+            )
+            raise ConfigurationError(
+                "Powerbi authorization failed . Please check your input configuration."
+            )
+
+        logger.info("Generated PowerBi access token")
+
+        self.__access_token = "Bearer {}".format(auth_response.get("access_token"))
+
+        logger.debug(f"{Constant.PBIAccessToken}={self.__access_token}")
+
+        return self.__access_token
+
+    def get_dashboards(self, workspace: Workspace) -> List[Dashboard]:
+        """
+        Get the list of dashboard from PowerBi for the given workspace identifier
+
+        TODO: Pagination. As per REST API doc (https://docs.microsoft.com/en-us/rest/api/power-bi/dashboards/get
+        -dashboards), there is no information available on pagination
+        """
+        dashboard_list_endpoint: str = self.get_dashboards_endpoint(workspace)
+
+        # Hit PowerBi
+        logger.info(f"Request to URL={dashboard_list_endpoint}")
+        response = self._request_session.get(
+            dashboard_list_endpoint,
+            headers={Constant.Authorization: self.get_access_token()},
+        )
+
+        response.raise_for_status()
+
+        dashboards_dict: List[Any] = response.json()[Constant.VALUE]
+
+        # Iterate through response and create a list of PowerBiAPI.Dashboard
+        dashboards: List[Dashboard] = [
+            Dashboard(
+                id=instance.get("id"),
+                isReadOnly=instance.get("isReadOnly"),
+                displayName=instance.get("displayName"),
+                embedUrl=instance.get("embedUrl"),
+                webUrl=instance.get("webUrl"),
+                workspace_id=workspace.id,
+                workspace_name=workspace.name,
+                tiles=[],
+                users=[],
+                tags=[],
+            )
+            for instance in dashboards_dict
+            if instance is not None
+        ]
+
+        return dashboards
+
+    def get_groups(self) -> List[dict]:
+        group_endpoint = self.get_groups_endpoint()
+        params: dict = {"$top": self.TOP, "$skip": 0, "$filter": "type eq 'Workspace'"}
+
+        def fetch_page(page_number: int) -> dict:
+            params["$skip"] = self.TOP * page_number
+            logger.debug(f"Query parameters = {params}")
+            response = self._request_session.get(
+                group_endpoint,
+                headers={Constant.Authorization: self.get_access_token()},
+                params=params,
+            )
+            response.raise_for_status()
+            return response.json()
+
+        # Hit PowerBi
+        logger.info(f"Request to groups endpoint URL={group_endpoint}")
+        zeroth_page = fetch_page(0)
+        logger.debug(f"Page 0 = {zeroth_page}")
+        if zeroth_page.get("@odata.count") is None:
+            raise ValueError(
```

Comment:
Should this be a value error? 

Also, should we just warn and continue in this case instead of failing the full connector?

Reply:
done